src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -10,6 +11,12 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
+        public static IEnumerable<object[]> Encodings_TestData()
```

Comment:
Yes, then we should have an NegativeEncodingTests.cs or something like that which has the negative tests for all of the encodings and nothing related to positive tests, and then the individual UTFWhateverEncodingTests.cs files would each contain the positive tests for that specific encoding.